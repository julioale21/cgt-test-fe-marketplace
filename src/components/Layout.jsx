import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';

export function Layout() {
  const cartItemsCount = 3;

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
}
