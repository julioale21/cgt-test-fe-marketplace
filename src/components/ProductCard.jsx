import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
