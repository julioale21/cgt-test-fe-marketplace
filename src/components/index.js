export { ContainedButton } from "./buttons/ContainedButton";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { ProductList } from "./ProductList";
export {ProductCard} from "./ProductCard";
