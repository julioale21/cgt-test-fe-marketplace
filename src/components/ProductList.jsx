import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
