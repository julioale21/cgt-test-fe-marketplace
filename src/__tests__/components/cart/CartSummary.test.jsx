import React from 'react';
import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { CartSummary } from '../../../components';

describe('CartSummary', () => {
  const defaultProps = {
    subtotal: 99.99,
    shipping: 0,
    taxes: 8.99,
    finalTotal: 108.98,
    onCheckout: jest.fn(),
    onContinueShopping: jest.fn()
  };

  it('should match snapshot', () => {
    const props = { ...defaultProps };

    const { container } = render(<CartSummary {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should render all components with correct values', () => {
    const props = { ...defaultProps };

    render(<CartSummary {...props} />);

    expect(screen.getByText('Order Summary')).toBeInTheDocument();
    expect(screen.getByText('$99.99')).toBeInTheDocument();
    expect(screen.getByText('Free')).toBeInTheDocument();
    expect(screen.getByText('$8.99')).toBeInTheDocument();
    expect(screen.getByText('$108.98')).toBeInTheDocument();
    expect(screen.getByText('Continue to Checkout')).toBeInTheDocument();
    expect(screen.getByText('Continue Shopping')).toBeInTheDocument();
    expect(screen.getByText('Secure Checkout')).toBeInTheDocument();
    expect(screen.getByText('Free shipping on all orders')).toBeInTheDocument();
  });

  it('should display shipping cost when not free', () => {
    const props = { ...defaultProps, shipping: 5.99 };

    render(<CartSummary {...props} />);

    expect(screen.getByText('$5.99')).toBeInTheDocument();
  });

  it('should handle floating point values correctly', () => {
    const props = {
      ...defaultProps,
      subtotal: 99.999,
      taxes: 8.999,
      finalTotal: 108.999
    };

    render(<CartSummary {...props} />);

    expect(screen.getByText('$100.00')).toBeInTheDocument();
    expect(screen.getByText('$9.00')).toBeInTheDocument();
    expect(screen.getByText('$109.00')).toBeInTheDocument();
  });

  it('should handle very large numbers', () => {
    const props = {
      ...defaultProps,
      subtotal: 999999.99,
      taxes: 99999.99,
      finalTotal: 1099999.98
    };

    render(<CartSummary {...props} />);

    const subtotalRow = screen.getByText('Subtotal').closest('div');
    const taxRow = screen.getByText('Estimated tax').closest('div');
    const totalRow = screen.getByText('Total').closest('div');

    expect(within(subtotalRow).getByText('$999,999.99')).toBeInTheDocument();
    expect(within(taxRow).getByText('$99,999.99')).toBeInTheDocument();
    expect(within(totalRow).getByText('$1,099,999.98')).toBeInTheDocument();
  });

  it('should handle non-positive values as zero', () => {
    const props = {
      ...defaultProps,
      subtotal: -99.99,
      taxes: -8.99,
      finalTotal: -108.98
    };

    render(<CartSummary {...props} />);

    const subtotalRow = screen.getByText('Subtotal').closest('div');
    const taxRow = screen.getByText('Estimated tax').closest('div');
    const totalRow = screen.getByText('Total').closest('div');

    expect(within(subtotalRow).getByText('$0.00')).toBeInTheDocument();
    expect(within(taxRow).getByText('$0.00')).toBeInTheDocument();
    expect(within(totalRow).getByText('$0.00')).toBeInTheDocument();
  });

  it('should call onCheckout when checkout button is clicked', async () => {
    const props = { ...defaultProps };
    const user = userEvent.setup();

    render(<CartSummary {...props} />);
    await user.click(screen.getByText('Continue to Checkout'));

    expect(props.onCheckout).toHaveBeenCalledTimes(1);
  });

  it('should handle undefined callback props', async () => {
    const props = {
      ...defaultProps,
      onCheckout: undefined,
      onContinueShopping: undefined
    };
    const user = userEvent.setup();

    render(<CartSummary {...props} />);

    await user.click(screen.getByText('Continue to Checkout'));
    await user.click(screen.getByText('Continue Shopping'));

    expect(screen.getByText('Continue to Checkout')).toBeInTheDocument();
  });

  it('should call onContinueShopping when continue shopping button is clicked', async () => {
    const props = { ...defaultProps };
    const user = userEvent.setup();

    render(<CartSummary {...props} />);
    await user.click(screen.getByText('Continue Shopping'));

    expect(props.onContinueShopping).toHaveBeenCalledTimes(1);
  });

  it('should handle zero values correctly', () => {
    const props = {
      ...defaultProps,
      subtotal: 0,
      taxes: 0,
      finalTotal: 0
    };

    render(<CartSummary {...props} />);

    const subtotalRow = screen.getByText('Subtotal').closest('div');
    const taxRow = screen.getByText('Estimated tax').closest('div');
    const totalRow = screen.getByText('Total').closest('div');
    expect(within(subtotalRow).getByText('$0.00')).toBeInTheDocument();
    expect(within(taxRow).getByText('$0.00')).toBeInTheDocument();
    expect(within(totalRow).getByText('$0.00')).toBeInTheDocument();
  });

  it('should render all payment method icons with correct attributes', () => {
    const props = { ...defaultProps };

    render(<CartSummary {...props} />);

    const paymentIcons = screen.getAllByRole('img');
    expect(paymentIcons).toHaveLength(3);
    paymentIcons.forEach((icon) => {
      expect(icon).toHaveAttribute('src');
      expect(icon).toHaveAttribute('alt');
    });
    expect(screen.getByAltText('Visa')).toBeInTheDocument();
    expect(screen.getByAltText('Mastercard')).toBeInTheDocument();
    expect(screen.getByAltText('PayPal')).toBeInTheDocument();
  });

  it('should maintain accessible structure with correct HTML tags', () => {
    const props = { ...defaultProps };

    render(<CartSummary {...props} />);

    expect(screen.getByText('Order Summary').tagName).toBe('SPAN');
    expect(screen.getByText('Secure Checkout').tagName).toBe('H6');
  });
});
