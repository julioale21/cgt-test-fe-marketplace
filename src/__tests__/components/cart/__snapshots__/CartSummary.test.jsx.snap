// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartSummary should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-yvg4p4"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-19hh0f1-MuiTypography-root"
    >
      Order Summary
    </h5>
    <div
      class="MuiStack-root css-1sazv7p-MuiStack-root"
    >
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
        >
          Subtotal
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          $99.99
        </p>
      </div>
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
        >
          Shipping
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-wl91uy-MuiTypography-root"
        >
          Free
        </p>
      </div>
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
        >
          Estimated tax
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          $8.99
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          Total
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          $108.98
        </h6>
      </div>
    </div>
    <div
      class="MuiStack-root css-1wviqzt-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-14lfj9a-MuiButtonBase-root-MuiButton-root"
        id=":r0:"
        tabindex="0"
        type="button"
      >
        Continue to Checkout
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary css-fdnlxf-MuiButtonBase-root-MuiButton-root"
        id=":r1:"
        tabindex="0"
        type="button"
      >
        Continue Shopping
      </button>
    </div>
    <div
      class="MuiBox-root css-4sywta"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-19b5ozw-MuiTypography-root"
      >
        Secure Checkout
      </p>
      <div
        class="MuiStack-root css-hbbo2v-MuiStack-root"
      >
        <img
          alt="Visa"
          class="MuiBox-root css-tv7otk"
          src="https://cdn.visa.com/v2/assets/images/logos/visa/blue/logo.png"
        />
        <img
          alt="Mastercard"
          class="MuiBox-root css-tv7otk"
          src="https://cdn.worldvectorlogo.com/logos/mastercard-7.svg"
        />
        <img
          alt="PayPal"
          class="MuiBox-root css-tv7otk"
          src="https://www.paypalobjects.com/digitalassets/c/website/logo/full-text/pp_fc_hl.svg"
        />
      </div>
      <div
        class="MuiStack-root css-xqxc1w-MuiStack-root"
      >
        <svg
          class="MuiBox-root css-skhuem"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8H4M20 16H4"
            fill="currentColor"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
        >
          Free shipping on all orders
        </p>
      </div>
    </div>
  </div>
</div>
`;
